lib/migration.js

```diff
@@ -42,11 +42,13 @@ function mixinMigration(MySQL, mysql) {
       var table = self.tableEscaped(model);
       self.execute('SHOW FIELDS FROM ' + table, function(err, fields) {
         self.execute('SHOW INDEXES FROM ' + table, function(err, indexes) {
-          if (!err && fields && fields.length) {
-            self.alterTable(model, fields, indexes, done);
-          } else {
-            self.createTable(model, done);
-          }
+          self.discoverForeignKeys(self.table(model), {}, function(discoverErr, foreignKeys) {
```

Comment:
> How should a foreign key discovery error be handled here?

Honestly, I don't know, because I am not familiar with this code base. Is it a bad idea to abort the whole autoupdate/automigrate operation in such case? If not, then a warning/error should be printed to the console at least.